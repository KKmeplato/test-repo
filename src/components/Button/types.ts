import type { VariantProps } from "class-variance-authority";
import { ButtonHTMLAttributes } from "react";

import { Size, Variant } from "lib/index";

import { buttonVariants } from "./utils";

export type ButtonVariantProps = (props?: {
  variant?: Exclude<Variant, "default" | "attention" | "info">;
  size?: Size;
  className: string | undefined;
}) => string;

export interface ButtonProps
  extends ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  icon?: React.ReactNode;
  disclosure?: "up" | "down";
  plain?: boolean;
  outline?: boolean;
  accessibilityLabel?: string;
  loading?: boolean;
}
