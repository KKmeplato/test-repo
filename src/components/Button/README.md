## Button

A button is a versatile UI component that allows users to take actions, make choices, and submit data. Our buttons are available in various sizes and versions, each designed to provide the optimal interaction experience across the platform.

### When to use

* When you want users to initiate or commit to an action like `Save`, `Submit`, `Delete`, etc.
* As a call-to-action, leading users to another part of the application or webpage.
* For confirming user decisions or choices.

```tsx
<Button 
  variant="primary" 
  size="small" 
  icon={<Icon.info />}
>
  Additional Information
</Button>
```

- [x] Meplato light theme
- [x] Meplato dark theme
- [x] Webridge light theme
- [x] Webridge dark theme