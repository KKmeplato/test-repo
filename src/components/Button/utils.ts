import { cva } from "class-variance-authority";

import { ButtonVariantProps } from "./types";

export const buttonVariants: ButtonVariantProps = cva(
  "rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1",
  {
    variants: {
      variant: {
        basic:
          "border border-default hover:border-hovered active:border-pressed text disabled:text-disabled bg-interactive-plain hover:bg-interactive-plain-hovered active:bg-interactive-plain-pressed disabled:bg-interactive-disabled disabled:border-disabled",
        primary:
          "bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled",
        critical:
          "bg-interactive-critical text-on-interactive hover:bg-interactive-critical-hovered active:bg-interactive-critical-pressed disabled:text-disabled disabled:bg-interactive-disabled",
        success:
          "bg-interactive-success text-on-interactive hover:bg-interactive-success-hovered active:bg-interactive-success-pressed disabled:text-disabled disabled:bg-interactive-disabled",
        outline: "",
        warning: "",
        highlight: ""
      },
      size: {
        small: "h-7 px-3 py-1",
        medium: "h-9 px-4 py-2",
        large: "h-11 px-6 py-3"
      }
    },
    defaultVariants: {
      variant: "basic",
      size: "medium"
    }
  }
);

export const plainButtonVariants: ButtonVariantProps = cva(
  "rounded-interactive-elements-default justify-center items-center gap-1 inline-flex leading-5 font-medium enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1",
  {
    variants: {
      variant: {
        basic:
          "text disabled:text-disabled hover:bg-surface-neutral-hovered active:bg-surface-neutral-pressed disabled:hover:bg-inherit",
        primary:
          "text-interactive-primary hover:text-interactive-primary-hovered active:text-interactive-primary-pressed disabled:text-disabled hover:bg-surface-selected-hovered active:bg-surface-selected-pressed disabled:hover:bg-inherit",
        critical:
          "text-interactive-critical hover:text-interactive-critical-hovered active:text-interactive-critical-pressed disabled:text-disabled hover:bg-surface-critical-hovered active:bg-surface-critical-pressed disabled:hover:bg-inherit",
        success:
          "text-interactive-success hover:text-interactive-success-hovered active:text-interactive-success-pressed disabled:text-disabled hover:bg-surface-success-hovered active:bg-surface-success-pressed disabled:hover:bg-inherit",
        warning:
          "text-interactive-warning hover:text-interactive-warning-hovered active:text-interactive-warning-pressed disabled:text-disabled hover:bg-surface-warning-hovered active:bg-surface-warning-pressed disabled:hover:bg-inherit",
        highlight:
          "text-interactive-highlight hover:text-interactive-highlight-hovered active:text-interactive-highlight-pressed disabled:text-disabled hover:bg-surface-highlight-hovered active:bg-surface-highlight-pressed disabled:hover:bg-inherit"
      },
      size: {
        small: "px-2 py-0.5 text-sm",
        medium: "px-2 py-0.5 text-sm",
        large: "px-2 py-0.5 text-base"
      }
    },
    defaultVariants: {
      variant: "basic",
      size: "small"
    }
  }
);

export const outlineButtonVariants: ButtonVariantProps = cva(
  "border rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm leading-5 font-medium enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1",
  {
    variants: {
      variant: {
        basic:
          "border-default text hover:border-hovered active:border-pressed disabled:text-disabled",
        primary:
          "text-interactive-primary border-interactive-primary hover:border-interactive-primary-hovered active:border-interactive-primary-pressed hover:text-interactive-primary-hovered active:text-interactive-primary-pressed disabled:border-disabled disabled:text-disabled",
        critical:
          "text-interactive-critical border-interactive-critical hover:border-interactive-critical-hovered active:border-interactive-critical-pressed hover:text-interactive-critical-hovered active:text-interactive-critical-pressed disabled:border-disabled disabled:text-disabled",
        success:
          "text-interactive-success border-interactive-success hover:border-interactive-success-hovered active:border-interactive-success-pressed hover:text-interactive-success-hovered active:text-interactive-success-pressed disabled:border-disabled disabled:text-disabled",
        warning:
          "text-interactive-warning border-interactive-warning hover:border-interactive-warning-hovered active:border-interactive-warning-pressed hover:text-interactive-warning-hovered active:text-interactive-warning-pressed disabled:border-disabled disabled:text-disabled",
        highlight:
          "text-interactive-highlight border-interactive-highlight hover:border-interactive-highlight-hovered active:border-interactive-highlight-pressed hover:text-interactive-highlight-hovered active:text-interactive-highlight-pressed disabled:border-disabled disabled:text-disabled"
      },
      size: {
        small: "py-1 px-3",
        medium: "py-1 px-3",
        large: "py-3 px-6"
      }
    },
    defaultVariants: {
      variant: "basic",
      size: "medium"
    }
  }
);
