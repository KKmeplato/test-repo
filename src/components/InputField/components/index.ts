export * from "./StepperControl";
export * from "./StepperIcon";
