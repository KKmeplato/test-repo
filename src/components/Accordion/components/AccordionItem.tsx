import * as RadixAccordion from "@radix-ui/react-accordion";
import { forwardRef } from "react";
import { useId } from "react";

import { AccordionItemProps } from "components/Accordion/types";
import { Button } from "components/Button";
import { Icon } from "components/icons";
import { cn } from "lib/utils";

export const AccordionItem = forwardRef<
  React.ElementRef<typeof RadixAccordion.Item>,
  AccordionItemProps
>(
  (
    { actionText, children, disabled, id, header, onActionClick, ...rest },
    forwardRef
  ) => {
    const generatedId = useId();
    const itemId = id || generatedId;

    return (
      <RadixAccordion.Item
        className={cn(
          "w-full rounded-level-2 hover:bg-surface-hovered focus:outline-none",
          "radix-disabled:bg-surface-disabled radix-disabled:hover:bg-surface-disabled radix-disabled:active:bg-surface-disabled",
          "[&:has([data-radix-collection-item]:active:not([data-disabled]))]:bg-surface-pressed "
        )}
        disabled={disabled}
        id={itemId}
        ref={forwardRef}
        {...rest}
      >
        <RadixAccordion.Header
          className={cn(
            "inline-flex w-full items-start justify-start gap-2",
            "radix-state-closed:rounded-level-2 radix-state-open:rounded-t-lg"
          )}
        >
          <RadixAccordion.Trigger className="group inline-flex shrink grow basis-0 gap-2 p-2 text-left focus:outline-none">
            <Icon.chevron
              className={cn(
                "h-5 w-5 shrink-0 p-0.5 text-icon ease-in-out motion-reduce:transition-none",
                "group-radix-state-closed:duration-300 group-radix-state-open:rotate-90 group-radix-state-open:duration-300",
                "group-focus-within:rounded group-focus-within:outline group-focus-within:outline-1 group-focus-within:outline-focus group-focus-within:ring-1 group-focus-within:ring-focus group-focus-within:ring-offset-1"
              )}
            />
            <span className="inline-flex shrink grow basis-0 flex-col items-start justify-center gap-2.5 rounded py-1 text-xs leading-none">
              {header}
            </span>
          </RadixAccordion.Trigger>
          {actionText && !disabled && (
            <div className="flex items-center justify-center rounded p-2">
              <Button
                disabled={disabled}
                plain
                variant="primary"
                size="small"
                onClick={onActionClick}
              >
                {actionText}
              </Button>
            </div>
          )}
        </RadixAccordion.Header>
        <RadixAccordion.Content className="rounded-b-lg p-2 pl-4 text-xs text radix-state-closed:animate-accordion-close radix-state-open:animate-accordion-open motion-reduce:transition-none">
          {children}
        </RadixAccordion.Content>
      </RadixAccordion.Item>
    );
  }
);

AccordionItem.displayName = "AccordionItem";
