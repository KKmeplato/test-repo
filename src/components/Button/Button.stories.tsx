import { action } from "@storybook/addon-actions";
import { Meta, StoryObj } from "@storybook/react";

import { Button } from "components/Button/index";
import { Icon } from "components/icons";

const meta: Meta<typeof Button> = {
  component: Button
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Default: Story = {
  args: {
    children: "Label",
    variant: "primary",
    icon: <Icon.info />,
    size: "small",
    id: "test-button-light-meplato",
    name: "button-light-meplato",
    disabled: false,
    disclosure: undefined,
    onClick: action("onClick")
  }
};
