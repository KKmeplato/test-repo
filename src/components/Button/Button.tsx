import { forwardRef } from "react";

import { Icon } from "components/icons";
import { cn } from "lib/utils";

import { ButtonProps } from "./types";
import {
  buttonVariants,
  outlineButtonVariants,
  plainButtonVariants
} from "./utils";

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      children,
      variant = "basic",
      plain = false,
      outline = false,
      size = "medium",
      disabled,
      className,
      icon,
      disclosure,
      loading = false,
      accessibilityLabel,
      onClick,
      ...rest
    },
    forwardedRef
  ) => {
    let buttonClasses = "";
    if (plain) {
      buttonClasses = plainButtonVariants({ variant, size, className });
    } else if (outline) {
      buttonClasses = outlineButtonVariants({ variant, size, className });
    } else {
      buttonClasses = buttonVariants({ variant, size, className });
    }

    return (
      <button
        data-size={size}
        data-variant={variant}
        data-plain={plain}
        data-outline={outline}
        className={cn(
          buttonClasses,
          {
            "disabled:text-transparent": loading
          },
          {
            "pr-2": disclosure && size === "small" && !plain,
            "pr-3": disclosure && size === "medium" && !plain,
            "pr-5": disclosure && size === "large" && !plain
          },
          {
            "pl-2": icon && size === "small" && !plain,
            "pl-3": icon && size === "medium" && !plain,
            "pl-5": icon && size === "large" && !plain
          },
          className
        )}
        ref={forwardedRef}
        disabled={disabled || loading}
        onClick={onClick}
        aria-busy={loading || undefined}
        aria-disabled={loading || disabled}
        {...rest}
      >
        {loading && (
          <Icon.loading
            className={
              "absolute inline-flex h-5 w-5 animate-spin flex-col items-center justify-center rounded-sm text-interactive-highlight"
            }
          />
        )}
        <>
          {icon && (
            <span
              className={cn(
                "inline-flex h-5 w-5 flex-col items-center justify-center rounded-sm",
                {
                  "text-icon-disabled": disabled,
                  "text-transparent": loading
                }
              )}
            >
              {icon}
            </span>
          )}
          {children}
          {disclosure && (
            <Icon.sort
              className={cn(
                "inline-flex h-5 w-5 flex-col items-center justify-center rounded-sm p-0.5",
                {
                  "rotate-180": disclosure === "up"
                }
              )}
            />
          )}
          {!children && accessibilityLabel && (
            <span className="sr-only">{accessibilityLabel}</span>
          )}
        </>
      </button>
    );
  }
);

Button.displayName = "Button";
