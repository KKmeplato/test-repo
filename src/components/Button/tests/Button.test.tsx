import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Button } from "components/Button";
import { Icon } from "components/icons";
import { Size } from "lib/index";

describe("<Button />", () => {
  interface Props {
    variant: "basic" | "primary" | "critical" | "success";
    size?: Size;
    disabled?: boolean;
    icon?: React.ReactNode;
    disclosure?: "up" | "down";
    testId?: string;
    loading?: boolean;
    onClick: () => void;
  }

  test("renders and re-renders with specified props", async () => {
    const { rerender } = render(
      <Button
        variant="primary"
        data-testid="test-button"
        accessibilityLabel="Click me"
        onClick={vi.fn()}
      >
        Get started
      </Button>
    );

    const buttonElement = await screen.findByTestId("test-button");
    expect(buttonElement).toBeInTheDocument();
    expect(within(buttonElement).queryByText("Get started")).not.toBeNull();

    rerender(
      <Button
        variant="critical"
        data-testid="test-button"
        disclosure="down"
        accessibilityLabel="Click me"
        onClick={vi.fn()}
      >
        Subscribe now
      </Button>
    );

    expect(buttonElement).toBeInTheDocument();
    expect(within(buttonElement).queryByText("Subscribe now")).not.toBeNull();
    /** Check if the button contains an SVG element */
    const svgIcon = buttonElement.querySelector("svg");
    expect(svgIcon).toBeInTheDocument();
  });

  test("renders with an after icon and invokes icon onClick callback", async () => {
    const mockButtonClick = vi.fn() as Props["onClick"];

    render(
      <Button
        variant="critical"
        data-testid="test-button"
        disclosure="down"
        accessibilityLabel="Click me"
        icon={<Icon.sort />}
        onClick={mockButtonClick}
      >
        Choose
      </Button>
    );

    /** Invokes onClick callback on button click */
    const buttonElement = await screen.findByTestId("test-button");
    expect(within(buttonElement).queryByText("Choose")).not.toBeNull();
    await userEvent.click(buttonElement);
    expect(mockButtonClick).toHaveBeenCalled();

    /** Invokes onClick callback on before icon click */
    const beforeSvgIcon = await buttonElement.querySelector("svg");
    expect(beforeSvgIcon).toBeInTheDocument();
    if (beforeSvgIcon) {
      userEvent.click(beforeSvgIcon);
      expect(mockButtonClick).toHaveBeenCalled();
    }

    /** Invokes onClick callback on after icon click */
    const afterSvgIcon = await buttonElement.querySelector("svg");
    expect(afterSvgIcon).toBeInTheDocument();
    if (afterSvgIcon) {
      userEvent.click(afterSvgIcon);
      expect(mockButtonClick).toHaveBeenCalled();
    }
  });

  test("renders disabled button", async () => {
    const mockButtonClick = vi.fn() as Props["onClick"];

    const { rerender } = render(
      <Button
        variant="primary"
        data-testid="test-button"
        accessibilityLabel="Click me"
        onClick={mockButtonClick}
      >
        Get started
      </Button>
    );

    const buttonElement = await screen.findByTestId("test-button");
    expect(within(buttonElement).queryByText("Get started")).not.toBeNull();
    expect(buttonElement).not.toBeDisabled();

    rerender(
      <Button
        variant="critical"
        data-testid="test-button"
        accessibilityLabel="Click me"
        disclosure="up"
        onClick={mockButtonClick}
        disabled
      >
        Subscribe now
      </Button>
    );

    expect(within(buttonElement).queryByText("Subscribe now")).not.toBeNull();
    expect(buttonElement).toBeDisabled();
  });

  test("renders correctly", () => {
    const componentWithChildren = render(
      <Button
        variant="primary"
        data-testid="test-button"
        accessibilityLabel="Click me"
        onClick={vi.fn()}
      >
        Save
      </Button>
    );
    expect(componentWithChildren).toMatchSnapshot();

    const component = render(
      <Button
        variant="primary"
        data-testid="test-button"
        accessibilityLabel="Click me"
        onClick={vi.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
