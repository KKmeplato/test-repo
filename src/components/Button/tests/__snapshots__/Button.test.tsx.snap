// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled h-9 px-4 py-2"
        data-outline="false"
        data-plain="false"
        data-size="medium"
        data-testid="test-button"
        data-variant="primary"
      >
        Save
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled h-9 px-4 py-2"
      data-outline="false"
      data-plain="false"
      data-size="medium"
      data-testid="test-button"
      data-variant="primary"
    >
      Save
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Button /> > renders correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled h-9 px-4 py-2"
        data-outline="false"
        data-plain="false"
        data-size="medium"
        data-testid="test-button"
        data-variant="primary"
      >
        Save
      </button>
    </div>
    <div>
      <button
        class="rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled h-9 px-4 py-2"
        data-outline="false"
        data-plain="false"
        data-size="medium"
        data-testid="test-button"
        data-variant="primary"
      >
        <span
          class="sr-only"
        >
          Click me
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="rounded-interactive-elements-default justify-center items-center gap-1 inline-flex text-sm font-medium leading-tight enabled:focus-visible:outline-focus enabled:focus-visible:outline-1 enabled:focus-visible:outline enabled:focus-visible:ring-focus enabled:focus-visible:ring-1 enabled:focus-visible:ring-offset-1 bg-primary text-on-interactive hover:bg-interactive-primary-hovered active:bg-interactive-primary-pressed disabled:text-disabled disabled:bg-interactive-disabled h-9 px-4 py-2"
      data-outline="false"
      data-plain="false"
      data-size="medium"
      data-testid="test-button"
      data-variant="primary"
    >
      <span
        class="sr-only"
      >
        Click me
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
